### ISSUE TITLE
VelocityInlet, MovingWall and DirichletField hold their values in static fields, so gradients through them are silently zero

### ISSUE THREAD

**Issue by ctdegroot** [DeGrootResearchGroup/aquaflux#363]
## Three boundary values are static, so a gradient through them is silently zero

```
aquaflux/flow/boundary.py:428   VelocityInlet.velocity:  object = eqx.field(static=True)
aquaflux/flow/boundary.py:377   MovingWall.velocity:     object = eqx.field(static=True)
aquaflux/boundary/conditions.py:139  DirichletField.field_fn: object = eqx.field(static=True)
```

Each class's only field is the static one, so instances have **zero pytree leaves**. `jax.grad`
with respect to an inlet velocity therefore returns **0.0, not NaN** -- the severed-adjoint failure
that a finiteness assertion cannot see, and that this project's testing guidance calls out
specifically.

This is not hypothetical scope: inlet velocity is the most obvious design variable in any flow
optimization, and `DirichletField` is how a sub-patch species injection is imposed, so an injection
profile's parameters are unreachable too.

## Why they are static, and why that reason does not cover the whole field

The stated reason is real: `_prescribed_components` (`flow/boundary.py:126-137`) keeps each
component a concrete scalar because "a jnp array here would not concretize under jit", and the
field must also accept a *callable* for a profile. So the **shape** of the closure -- a constant
vector versus a callable versus which profile family -- genuinely belongs on the static side.

The **coefficients** do not. A parabolic inlet's peak velocity, a raised-cosine injector's centre
and width, a moving wall's speed: these are ordinary numbers that want to be leaves.

## Proposed direction

Split the two concerns rather than flipping the flag. A small `Profile` module whose *structure* is
static and whose *coefficients* are array leaves:

- static: which profile, its axes, anything that changes shapes or control flow;
- dynamic: the numbers it evaluates with.

`VelocityInlet.velocity` then holds a `Profile` (a uniform vector being the trivial one), the jit
concretization argument is satisfied by the static half, and `jax.grad` reaches the coefficients.

Worth checking as part of this: five of the nine closures already carry dynamic leaves --
`Dirichlet.value`, `Neumann.flux`, `Convective.h` / `.t_inf`, `PressureOutlet.pressure` -- so the
gradient path is real and only these three are severed. Any fix should leave those untouched.

## A related trap for whoever builds on this

Those five working leaves are annotated `float`. A Python float assigned to them is filtered out by
`equinox.filter_grad`, so a caller that passes a plain number instead of `jnp.asarray(...)`
reproduces the same silent zero on the closures that otherwise work. Whatever constructs these from
user input has to build arrays.

## Tests

Per the project's guidance that "without NaNs" is the floor rather than the test: compare the
gradient against a finite difference where cost allows, and at minimum assert it is **non-zero** --
a test named "is differentiable in X" that only asserts finiteness passes on a severed adjoint.
